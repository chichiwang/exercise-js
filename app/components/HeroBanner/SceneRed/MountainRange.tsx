import React from 'react';

import styles from './styles.mountainrange.module.scss';

function MountainRange() {
  return (
    <div className={styles.mountainRange}>
      <div className={styles.m01} />
      <div className={styles.m02} />
      <div className={styles.m03} />
      <div className={styles.m04} />
      <div className={styles.m05} />
      <div className={styles.m06} />
      <div className={styles.m07} />
      <div className={styles.m08} />
      <div className={styles.m09} />
      <div className={styles.m10} />
      <div className={styles.m11} />
      <div className={styles.m12} />
      <div className={styles.m13} />
    </div>
  );
}

export default MountainRange;
