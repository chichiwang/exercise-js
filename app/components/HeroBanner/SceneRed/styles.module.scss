@use "sass:map";
@use "app/var.module";

$gap-height: 3px;

.field {
  position: relative;
  width: 100%;
  height: 7.5px;
  background-color: map.get(var.$colors, 'red');

  &::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: map.get(var.$colors, 'red');
  }

  .scene {
    margin-left: 0px;
    position: absolute;
    width: 10px;
    height: 100%;
    top: 0%;
    left: 50%;
    transform: translate(-50%, 0%);
  }
}
