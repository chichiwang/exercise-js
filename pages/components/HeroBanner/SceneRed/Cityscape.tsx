import React from 'react';
import classNames from 'classnames';

import styles from './styles.module.scss';

function Cityscape() {
  return (
    <>
      <div className={styles.fiveBuilding}>
        <div className={styles.base} />
        <div className={classNames(styles.level, styles.one)} />
        <div className={classNames(styles.level, styles.two)} />
        <div className={classNames(styles.level, styles.three)} />
        <div className={classNames(styles.level, styles.four)} />
        <div className={classNames(styles.level, styles.five)} />
        <div className={classNames(styles.level, styles.six)} />
        <div className={classNames(styles.level, styles.seven)} />
        <div className={classNames(styles.level, styles.eight)} />
        <div className={styles.topBase} />
        <div className={styles.antenna} />
        <div className={styles.sign} />
      </div>

      <div className={styles.cappedBldg} />
      <div className={classNames(styles.cappedBldg, styles.bldg01)} />
      <div className={classNames(styles.cappedBldg, styles.bldg02)} />
      <div className={styles.flatBldg} />
      <div className={classNames(styles.flatBldg, styles.bldg01)} />
      <div className={classNames(styles.flatBldg, styles.bldg02)} />
      <div className={styles.domedBldg} />
      <div className={classNames(styles.domedBldg, styles.bldg01)} />
    </>
  );
}

export default Cityscape;
