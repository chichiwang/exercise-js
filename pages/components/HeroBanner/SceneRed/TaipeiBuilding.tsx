import React from 'react';
import classNames from 'classnames';

import styles from './styles.module.scss';

function TaipeiBuilding() {
  return (
    <div className={styles.taipei101}>
      <div className={styles.base} />
      <div className={classNames(styles.level, styles.one)} />
      <div className={classNames(styles.level, styles.two)} />
      <div className={classNames(styles.level, styles.three)} />
      <div className={classNames(styles.level, styles.four)} />
      <div className={classNames(styles.level, styles.five)} />
      <div className={classNames(styles.level, styles.six)} />
      <div className={classNames(styles.level, styles.seven)} />
      <div className={classNames(styles.level, styles.eight)} />
      <div className={styles.topBase} />
      <div className={styles.antenna} />
      <div className={styles.sign} />
    </div>
  );
}

export default TaipeiBuilding;
