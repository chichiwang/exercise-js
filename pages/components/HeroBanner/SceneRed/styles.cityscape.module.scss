@use "sass:map";
@use "app/var.module";

$gap-height: 3px;

.cityscape {
  position: absolute;
  left: -582px;
  bottom: 0;
  width: 400px;
  height: 295px;

  .cappedBldg {
    position: absolute;
    bottom: 0;
    left: 244px;
    width: 45px;
    height: 78px;
    background-color: map.get(var.$colors, 'red');
    border-top: $gap-height solid map.get(var.$colors, 'white');
    border-right: $gap-height solid map.get(var.$colors, 'white');
    border-left: $gap-height solid map.get(var.$colors, 'white');

    &::before {
      content: '';
      position: absolute;
      width: 39px;
      height: 8px;
      left: 0px;
      top: -8px;
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
    }

    &.bldg01 {
      left: 0px;
      width: 60px;
      height: 52px;

      &::before {
        height: 12px;
        left: -3px;
        top: -12px;
      }
    }

    &.bldg02 {
      left: 323px;
      width: 42px;
      height: 64px;

      &::before {
        width: 34px;
        height: 22px;
        left: 5px;
        top: -22px;
      }
    }
  }

  .domedBldg {
    position: absolute;
    bottom: 0;
    left: 368px;
    width: 30px;
    height: 75px;
    background-color: map.get(var.$colors, 'red');
    border-top: $gap-height solid map.get(var.$colors, 'white');
    border-right: $gap-height solid map.get(var.$colors, 'white');
    border-left: $gap-height solid map.get(var.$colors, 'white');

    &::before {
      content: '';
      position: absolute;
      top: -12px;
      left: 1px;
      width: 22px;
      height: 22px;
      border-radius: 50%;
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
    }

    &::after {
      content: '';
      position: absolute;
      top: 0px;
      left: 0px;
      width: 100%;
      height: 100%;
      background-color: map.get(var.$colors, 'red');
    }

    &.bldg01 {
      left: 98px;
      width: 42px;
      height: 41px;

      &::before {
        top: -20px;
        width: 34px;
        height: 34px;
      }
    }
  }

  .flatBldg {
    position: absolute;
    bottom: 0;
    left: 291px;
    width: 30px;
    height: 50px;
    background-color: map.get(var.$colors, 'red');
    border-top: $gap-height solid map.get(var.$colors, 'white');
    border-right: $gap-height solid map.get(var.$colors, 'white');
    border-left: $gap-height solid map.get(var.$colors, 'white');

    &.bldg01 {
      left: 64px;
      width: 32px;
      height: 72px;
    }

    &.bldg02 {
      left: 144px;
      width: 35px;
      height: 42px;
    }
  }

  .taipei101 {
    position: absolute;
    bottom: 0;
    left: 182px;
    width: 60px;
    height: 270px;

    & .base {
      position: absolute;
      bottom: 0;
      left: 0;
      width: 60%;
      height: 55px;

      &::before,
      &::after {
        content: '';
        position: absolute;
        bottom: 0;
        width: 100%;
        height: 100%;
        background-color: map.get(var.$colors, 'red');
        border-top: $gap-height solid map.get(var.$colors, 'white');
      }

      &::before {
        left: 0%;
        transform: skewX(-5deg);
        border-left: $gap-height solid map.get(var.$colors, 'white');
      }

      &::after {
        left: 66%;
        transform: skewX(5deg);
        border-right: $gap-height solid map.get(var.$colors, 'white');
      }
    }

    & .level {
      position: absolute;
      left: 0;
      width: 60%;
      height: 25px;

      &.one {
        bottom: 52px;
      }

      &.two {
        bottom: 74px;
      }

      &.three {
        bottom: 96px;
      }

      &.four {
        bottom: 118px;
      }

      &.five {
        bottom: 140px;
      }

      &.six {
        bottom: 162px;
      }

      &.seven {
        bottom: 184px;
      }

      &.eight {
        bottom: 206px;
      }

      &::before,
      &::after {
        content: '';
        position: absolute;
        bottom: 0;
        width: 100%;
        height: 100%;
        background-color: map.get(var.$colors, 'red');
        border-top: $gap-height solid map.get(var.$colors, 'white');
      }

      &::before {
        left: 10%;
        transform: skewX(8deg);
        border-left: $gap-height solid map.get(var.$colors, 'white');
      }

      &::after {
        left: 56%;
        transform: skewX(-8deg);
        border-right: $gap-height solid map.get(var.$colors, 'white');
      }
    }

    & .topBase {
      position: absolute;
      left: 17px;
      bottom: 228px;
      width: 40%;
      height: 10px;
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');

      &::before,
      &::after {
        content: '';
        position: absolute;
        top: -20px;
        width: 55%;
        height: 20px;
        background-color: map.get(var.$colors, 'red');
        border-top: $gap-height solid map.get(var.$colors, 'white');
      }

      &::before {
        left: 1px;
        transform: skewX(6deg);
        border-left: $gap-height solid map.get(var.$colors, 'white');
      }

      &::after {
        left: 42%;
        transform: skewX(-6deg);
        border-right: $gap-height solid map.get(var.$colors, 'white');
      }
    }

    & .antenna {
      position: absolute;
      left: 23px;
      bottom: 252px;
      width: 20%;
      height: 8px;
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');

      &::before {
        content: '';
        position: absolute;
        left: 1px;
        bottom: 5px;
        width: 4px;
        height: 35px;
        background-color: map.get(var.$colors, 'white');
      }
    }

    & .sign {
      position: absolute;
      left: 19px;
      bottom: 40px;
      width: 20px;
      height: 20px;
      border-radius: 50%;
      border: 2px solid map.get(var.$colors, 'white');

      &::before {
        content: '';
        position: absolute;
        top: 4px;
        left: 4px;
        width: 50%;
        height: 50%;
        background-color: map.get(var.$colors, 'white');
      }
    }
  }
}
