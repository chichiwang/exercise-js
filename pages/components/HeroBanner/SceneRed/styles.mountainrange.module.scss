@use "sass:map";
@use "app/var.module";

$gap-height: 3px;

.mountainRange {
  position: absolute;
  left: 200px;
  bottom: 0;
  width: 550px;
  height: 150px;

  .m01 {
    position: absolute;
    left: 0;
    bottom: 0;
    width: 35px;
    height: 35px;

    &::before {
      content: '';
      position:absolute;
      top: 13px;
      left: 7px;
      width: 100%;
      height: 100%;
      transform: rotateZ(50deg);
      border-top-left-radius: 20% 25%;
      background-color: map.get(var.$colors, 'red');
      border: $gap-height solid map.get(var.$colors, 'white');
    }
  }
  .m02 {
    position: absolute;
    left: 35px;
    bottom: 0;
    width: 65px;
    height: 25px;

    &::before {
      content: '';
      position:absolute;
      top: -16px;
      left: 0px;
      width: 100%;
      height: 220%;
      transform: rotateZ(-30deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-top-right-radius: 20% 12%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 11px;
      left: -17px;
      width: 50%;
      height: 50%;
      transform: rotateZ(50deg);
      background-color: map.get(var.$colors, 'red');
      border-top-left-radius: 20% 20%;
    }
  }
  .m03 {
    position: absolute;
    left: 80px;
    bottom: 0;
    width: 95px;
    height: 55px;

    &::before {
      content: '';
      position:absolute;
      top: 18px;
      left: -27px;
      width: 112%;
      height: 150%;
      transform: rotateZ(25deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
      border-top-left-radius: 14% 15%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 5px;
      left: -22px;
      width: 29%;
      height: 99%;
      transform: rotateZ(25deg);
      background-color: map.get(var.$colors, 'red');
      border-top-left-radius: 60% 45%;
    }
  }
  .m04 {
    position: absolute;
    left: 120px;
    bottom: 0;
    width: 95px;
    height: 55px;

    &::before {
      content: '';
      position:absolute;
      top: 5px;
      left: -12px;
      width: 100%;
      height: 200%;
      transform: rotateZ(-45deg);
      background-color: map.get(var.$colors, 'red');
      border: $gap-height solid map.get(var.$colors, 'white');
      border-top-right-radius: 11% 11%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 32px;
      left: -35px;
      width: 100%;
      height: 100%;
      transform: rotateZ(25deg);
      background-color: map.get(var.$colors, 'red');
    }
  }
  .m05 {
    position: absolute;
    left: 134px;
    bottom: 0;
    width: 95px;
    height: 55px;

    &::before {
      content: '';
      position:absolute;
      top: 6px;
      left: -4px;
      width: 85%;
      height: 100%;
      transform: rotateZ(30deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-top-left-radius: 14% 20%;
    }
  }
  .m06 {
    position: absolute;
    left: 170px;
    bottom: 0;
    width: 72px;
    height: 100px;

    &::before {
      content: '';
      position:absolute;
      top: 5px;
      left: 10px;
      width: 128%;
      height: 100%;
      transform: rotateZ(48deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
      border-top-left-radius: 15% 12%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 51px;
      left: -23px;
      width: 50%;
      height: 50%;
      transform: rotateZ(30deg);
      background-color: map.get(var.$colors, 'red');
    }
  }
  .m07 {
    position: absolute;
    left: 240px;
    bottom: 0;
    width: 36px;
    height: 110px;

    &::before {
      content: '';
      position: absolute;
      left: 7px;
      top: -8px;
      width: 100%;
      height: 100%;
      transform: rotateZ(-52deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');
      border-top-right-radius: 20% 12%;
    }

    &::after {
      content: '';
      position: absolute;
      left: 1px;
      top: -3px;
      width: 100%;
      height: 100%;
      transform: rotateZ(-45deg);
      background-color: map.get(var.$colors, 'red');
      border-top-right-radius: 30% 15%;
    }
  }
  .m08 {
    position: absolute;
    left: 215px;
    bottom: 0;
    width: 80px;
    height: 95px;

    &::before {
      content: '';
      position:absolute;
      top: 0px;
      left: 48px;
      width: 112%;
      height: 171%;
      transform: rotateZ(-55deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');
      border-top-right-radius: 10% 5%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 10px;
      left: -9px;
      width: 100%;
      height: 100%;
      transform: rotateZ(36deg);
      background-color: map.get(var.$colors, 'red');
    }
  }
  .m09 {
    position: absolute;
    left: 330px;
    bottom: 0;
    width: 80px;
    height: 40px;

    &::before {
      content: '';
      position:absolute;
      top: 8px;
      left: 0px;
      width: 100%;
      height: 120%;
      transform: rotateZ(-15deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');
      border-top-right-radius: 15% 15%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 0px;
      left: -34px;
      width: 100%;
      height: 100%;
      transform: rotateZ(34deg);
      background-color: map.get(var.$colors, 'red');
    }
  }
  .m10 {
    position: absolute;
    left: 390px;
    bottom: 0;
    width: 80px;
    height: 40px;

    &::before {
      content: '';
      position:absolute;
      top: 11px;
      left: -5px;
      width: 100%;
      height: 100%;
      transform: rotateZ(20deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
      border-top-left-radius: 15% 15%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 7px;
      left: -24px;
      width: 50%;
      height: 100%;
      transform: rotateZ(-15deg);
      background-color: map.get(var.$colors, 'red');
      border-top-right-radius: 15% 15%;
    }
  }
  .m11 {
    position: absolute;
    left: 425px;
    bottom: 0;
    width: 80px;
    height: 30px;

    &::before {
      content: '';
      position:absolute;
      top: 8px;
      left: 0px;
      width: 100%;
      height: 100%;
      transform: rotateZ(10deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
    }
  }
  .m12 {
    position: absolute;
    left: 470px;
    bottom: 0;
    width: 50px;
    height: 20px;

    &::before {
      content: '';
      position:absolute;
      top: 0px;
      left: 0px;
      width: 100%;
      height: 150%;
      transform: rotateZ(-20deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-right: $gap-height solid map.get(var.$colors, 'white');
      border-top-right-radius: 15% 25%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 3px;
      left: -6px;
      width: 50%;
      height: 100%;
      transform: rotateZ(10deg);
      background-color: map.get(var.$colors, 'red');
    }
  }
  .m13 {
    position: absolute;
    left: 499px;
    bottom: 0;
    width: 40px;
    height: 20px;

    &::before {
      content: '';
      position:absolute;
      top: 0px;
      left: 0px;
      width: 120%;
      height: 150%;
      transform: rotateZ(22deg);
      background-color: map.get(var.$colors, 'red');
      border-top: $gap-height solid map.get(var.$colors, 'white');
      border-left: $gap-height solid map.get(var.$colors, 'white');
      border-top-left-radius: 15% 25%;
    }

    &::after {
      content: '';
      position:absolute;
      top: 0px;
      left: -6px;
      width: 50%;
      height: 100%;
      transform: rotateZ(-20deg);
      background-color: map.get(var.$colors, 'red');
    }
  }
}
