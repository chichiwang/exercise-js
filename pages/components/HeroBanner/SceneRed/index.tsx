import React from 'react';

import styles from './styles.module.scss';

import Cityscape from './Cityscape';

function SceneRed() {
  return (
    <div className={styles.field}>
      <div className={styles.scene}>
        <Cityscape />
      </div>
    </div>
  );
}

export default SceneRed;
