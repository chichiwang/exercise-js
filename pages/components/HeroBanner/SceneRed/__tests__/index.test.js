import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import SceneRed from '../index';

describe('HeroBanner - SceneRed', () => {
  test('matches snapshot', () => {
    const tree = renderer.create(<SceneRed />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
