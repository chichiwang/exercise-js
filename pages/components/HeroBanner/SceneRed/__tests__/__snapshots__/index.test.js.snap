// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner - SceneRed matches snapshot 1`] = `
<div
  className="field"
>
  <div
    className="scene"
  >
    <div
      className="fiveBuilding"
    >
      <div
        className="base"
      />
      <div
        className="level one"
      />
      <div
        className="level two"
      />
      <div
        className="level three"
      />
      <div
        className="level four"
      />
      <div
        className="level five"
      />
      <div
        className="level six"
      />
      <div
        className="level seven"
      />
      <div
        className="level eight"
      />
      <div
        className="topBase"
      />
      <div
        className="antenna"
      />
      <div
        className="sign"
      />
    </div>
    <div
      className="cappedBldg"
    />
    <div
      className="cappedBldg bldg01"
    />
    <div
      className="cappedBldg bldg02"
    />
    <div
      className="flatBldg"
    />
    <div
      className="flatBldg bldg01"
    />
    <div
      className="flatBldg bldg02"
    />
    <div
      className="domedBldg"
    />
    <div
      className="domedBldg bldg01"
    />
  </div>
</div>
`;
